File: 80-razno/jel-prost.js
const broj = process.argv[2] ? Number(process.argv[2]) : 35

function jelProst(broj) {
  if (broj < 2) return false
  if (broj == 2) return true
  if (broj % 2 == 0) return false
  for (let i = 3; i * i <= broj; i += 2)
    if (broj % i == 0) return false
  return true
}

console.log(jelProst(broj))

File: 80-razno/jel-prost.py
# https://stackoverflow.com/questions/1801391/what-is-the-best-algorithm-for-checking-if-a-number-is-prime

import sys
broj = int(sys.argv[1])

def jel_prost(n):
    if n == 2 or n == 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False

    i = 5
    w = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += w
        w = 6 - w

    return True


print(jel_prost(broj))
